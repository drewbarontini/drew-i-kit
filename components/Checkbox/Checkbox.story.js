import React from 'react';
import { storiesOf } from '@storybook/react';

import Checkbox from './';
import Flex, { FlexItem } from '../Flex';
import Text from '../Text';

storiesOf('Checkbox', module)
  .add('default', () => <Checkbox name="checkbox" />)
  .add('label', () => (
    <Flex alignItems="center">
      <FlexItem marginRight="xs">
        <Checkbox name="checkbox" />
      </FlexItem>
      <FlexItem>
        <Text as="label" htmlFor="checkbox" fontSize="s" fontWeight="bold">
          Label
        </Text>
      </FlexItem>
    </Flex>
  ));
