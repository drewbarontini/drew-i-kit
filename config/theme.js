const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
