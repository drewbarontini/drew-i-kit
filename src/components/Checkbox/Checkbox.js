import React from 'react';
import PropTypes from 'prop-types';

import StyledCheckbox from './styled/Checkbox';

const Checkbox = ({ name, ...rest }) => (
  <StyledCheckbox type="checkbox" name={name} id={name} {...rest} />
);

Checkbox.propTypes = {
  name: PropTypes.string.isRequired,
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
