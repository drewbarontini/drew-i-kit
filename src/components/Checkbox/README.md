# Checkbox

## Usage

```jsx
import { Checkbox } from 'drew-i-kit';

<Checkbox name="checkbox" />
```

## Props

- `name` (**string**)

_Also includes all default props for `<input>` elements._