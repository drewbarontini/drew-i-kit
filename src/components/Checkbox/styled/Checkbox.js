import styled from 'styled-components';

import { responsive } from '../../../lib';

const Checkbox = styled.input`
  ${responsive};
`;

Checkbox.displayName = 'StyledCheckbox';

export default Checkbox;
