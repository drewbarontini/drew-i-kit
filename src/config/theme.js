import colors from './colors';
import fonts from './fonts';
import icons from './icons';

const theme = {
  borders: {
    sizes: {
      base: '1px',
      l: '2px',
    },
    radius: {
      base: '3px',
    },
  },
  breakpoints: {
    sizes: {
      s: 380,
      m: 768,
      l: 900,
      xl: 1100,
    },
    query: 'max-width',
    unit: 'px',
  },
  colors: {
    ...colors,
  },
  fonts: {
    ...fonts,
  },
  icons: {
    ...icons,
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  sizing: {
    base: '900px',
    s: '400px',
    l: '1200px',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
