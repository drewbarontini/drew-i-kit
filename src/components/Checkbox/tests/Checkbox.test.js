import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import theme from '../../../config/theme';

import Checkbox from '..';

describe('<Checkbox />', () => {
  it('should render properly', () => {
    const { container } = render(<Checkbox theme={theme} name="checkbox" />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName.toLowerCase()).toEqual('input');
    expect(container.firstChild).toHaveAttribute('id');
    expect(container.firstChild.id).toEqual('checkbox');
    expect(container.firstChild).toHaveAttribute('name');
    expect(container.firstChild.name).toEqual('checkbox');
    expect(container.firstChild).toHaveAttribute('type');
    expect(container.firstChild.type).toEqual('checkbox');
  });
});
