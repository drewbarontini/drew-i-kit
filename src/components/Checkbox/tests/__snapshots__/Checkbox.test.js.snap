// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render properly 1`] = `
<input
  class=""
  id="checkbox"
  name="checkbox"
  type="checkbox"
/>
`;
